build_runner/test/server/serve_handler_test.dart

```diff
@@ -199,17 +199,14 @@ class MockWatchImpl implements WatchImpl {
   }
 
   MockWatchImpl(this.reader, this.packageGraph, this.assetGraph) {
+    var firstBuild = new Completer<BuildResult>();
```

Comment:
Couldn't you use Stream.forEach or something here?

I get nervous when I see Completer and fields holding onto Future